Exercise_1.java

```diff
@@ -1,8 +1,28 @@
+// Time Complexity : O(logn) (where n is the number of elements in the array)
+// Space Complexity : O(n) (array size)

```

Comment:
O(1) since we are only using 1 variable mid
if done recursively then space increases to lgn recursive stack space